### PR TITLE
Fixes #9: Show per-owner git hash in UI

### LINKED ISSUE #9: Show the git hash of the DAG in the Airflow UI
We had an issue where AWS sync did not pick up the newest files because the file size did not change but the contents changed. We want airflow to version control the DAGs (as referenced by the pr below) and then show the git hash in a column of the UI. This is for us to make sure that we ran the pipeline we expected to run with the correct config files.

### PR DESCRIPTION
**Pull request by gmarks-coatue**
Also see https://github.com/coatue/datasyncs/pull/594, the datasyncs PR providing the hashes. For appropriate owners, shows the most recent SHA-1 hash of the HEAD of that owner's repo and hyperlinks to the repo at that hash. For example, if the most recent revision of the datasyncs repo had SHA-1 hash `abc1234def...`, we would display "datasyncs (SHA: abc1234def)" and the SHA would link to `https://github.com/coatue/datasyncs/tree/abc1234def...`

Before:
Screen Shot 2019-08-07 at 1 50 55 PM

After: 
Screen Shot 2019-08-07 at 2 45 48 PM

### CODE REVIEW

File: airflow/www/views.py [hunk 1]
```diff
@@ -2194,6 +2200,37 @@ def get_int_arg(value, default=0):
...
+
+        owner_dict = {}
+        if REPO_SYNCS_CONFIG:
+            # for finding the sync_repos dag, we only care about DAGs with objects.
+            sync_repos_dags = list(filter(lambda dag: dag.dag_id == REPO_SYNCS_CONFIG.get('DEFAULT', 'DAG_ID'), dags))
+            assert len(sync_repos_dags) < 2, 'Duplicated dag_id in DAGs list!'
+            if sync_repos_dags:
+                sync_repos_tasks = dagbag.get_dag(sync_repos_dags[0].dag_id).tasks
+                task_id_prefix = REPO_SYNCS_CONFIG.get('DEFAULT', 'SHA_EXPORT_TASK_ID_PREFIX')
+                export_shas_tasks = filter(lambda task: task.task_id.startswith(task_id_prefix), sync_repos_tasks)
```
Comment (jsullivan-coatue) [inline airflow/www/views.py]: use list comprehension

File: airflow/www/views.py [hunk 2]
```diff
@@ -2194,6 +2200,37 @@ def get_int_arg(value, default=0):
...
+                repos = json.loads(REPO_SYNCS_CONFIG.get('DEFAULT', 'REPOS'))
+
+                for task in export_shas_tasks:
+                    repo_name = task.task_id[len(task_id_prefix) + 1:]
+                    for pair in repos:
+                        if repo_name == pair['repo']:
+                            break
+                    else:
```
Comment (jsullivan-coatue) [inline airflow/www/views.py]: is this indent correct?

Comment (gmarks-coatue) [reply airflow/www/views.py]: yes, it's a for-else so that if the task has an invalid repo name it is ignored

File: airflow/www/views.py [hunk 3]
```diff
@@ -2194,6 +2200,37 @@ def get_int_arg(value, default=0):
 
         max_dt = pendulum.timezone('UTC').convert(dt.datetime(9999, 12, 31))
         dags = list(sorted(obj_dags, key=lambda dag: dagbag.get_dag(dag.dag_id).get_next_run_date(pendulum.now('UTC')) or max_dt))
+
+        owner_dict = {}
+        if REPO_SYNCS_CONFIG:
+            # for finding the sync_repos dag, we only care about DAGs with objects.
+            sync_repos_dags = list(filter(lambda dag: dag.dag_id == REPO_SYNCS_CONFIG.get('DEFAULT', 'DAG_ID'), dags))
+            assert len(sync_repos_dags) < 2, 'Duplicated dag_id in DAGs list!'
```
Comment (pathikrit) [inline airflow/www/views.py]: prefer `<= 1` than `< 2` :)

File: airflow/www/views.py (hunk 2 above)
Comment (pathikrit) [reply airflow/www/views.py]: yup my favorite python feature that I probably won't recommend :)

File: airflow/www/views.py [hunk 4]
```diff
@@ -2194,6 +2200,37 @@ def get_int_arg(value, default=0):
...
+            if sync_repos_dags:
+                task_id_prefix = REPO_SYNCS_CONFIG.get('DEFAULT', 'SHA_EXPORT_TASK_ID_PREFIX')
+                sync_repos_tasks = dagbag.get_dag(sync_repos_dags[0].dag_id).tasks
+                export_shas_tasks = [task for task in sync_repos_tasks if task.task_id.startswith(task_id_prefix)]
+                repos = json.loads(REPO_SYNCS_CONFIG.get('DEFAULT', 'REPOS'))
+
+                for task in export_shas_tasks:
+                    repo_name = task.task_id[len(task_id_prefix) + 1:]
+                    for pair in repos:
```
Comment (pathikrit) [inline airflow/www/views.py]: code comment

**Pull request merged**